airflow/api_fastapi/views/public/dags.py

```diff
@@ -87,6 +93,34 @@ async def get_dags(
     )
 
 
+@dags_router.get(
+    "/dags/{dag_id}/details", responses=create_openapi_http_exception_doc([400, 401, 403, 404, 422])
+)
+async def get_dag_details(
+    dag_id: str, session: Annotated[Session, Depends(get_session)], request: Request
+) -> DAGDetailsResponse:
+    """Get details of DAG."""
+    if not session:
+        raise HTTPException(401, "Invalid session")
+
+    dag: DAG = request.app.state.dag_bag.get_dag(dag_id)
+    if not dag:
+        raise HTTPException(404, f"Dag with id {dag_id} was not found")
+
+    dag_model: DagModel = session.get(DagModel, dag_id)
+    if not dag_model:
+        raise HTTPException(400, f"Unable to obtain dag with id {dag_id} from session")
+
+    for key, value in dag.__dict__.items():
+        if not key.startswith("_") and not hasattr(dag_model, key):
+            setattr(dag_model, key, value)
+
+    try:
+        return DAGDetailsResponse.model_validate(dag_model, from_attributes=True)
+    except ValidationError as ve:
+        raise HTTPException(422, f"Error while validating dag model with id {dag_id}, details: {str(ve)}")
```

Comment:
FastAPI already has a native exception handler for pydantic due to his tight integration with it. FastAPI will take care of catching any `pydantic.ValidationError` and return the appropriate 422 response. (nicely formatted)
```suggestion
    return DAGDetailsResponse.model_validate(dag_model, from_attributes=True)
```

Reply:
Done.